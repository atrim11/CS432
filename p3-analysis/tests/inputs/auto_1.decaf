// Parameter mismatch, type errors, and recursion
def int multiply(int a, int b) {
    return a * b;
}

def int factorial(int n) {
    if (n <= 1) {
        return 1;
    }
    return n * factorial(n - 1);
}

def void printMessage(int msg) {
    print_int(msg);
}

def int add(int x, int y) {
    return x + y;
}
int arr[5];
def int main() {
    int result;
    int a;
    int b;
    int c;
    int y;
    int z;
    result = multiply(3);            // Error: multiply expects 2 parameters, provided 1
    result = multiply(5, "text");    // Error: second parameter should be an int
    result = add(4);                 // Error: add expects 2 parameters, provided 1
    result = add(5, "str");          // Error: second parameter should be int, not string

    a = 10;
    b = 20;
    c = a * b;
    printMessage(c);                 // Error: printMessage expects a string, provided an int
    
    // Recursive issue - bad base case
    result = factorial("text");      // Error: factorial expects an int, passed string
    result = factorial(-10);         // Edge case: what happens with negative numbers?

    // Type mismatches in function calls
    printMessage(2);               // Correct use
    printMessage(100);               // Error: printMessage expects a string, provided an int

    // Array errors
    arr[10] = 100;                   // Error: array index out of bounds
    y = arr[2] + 15;             // Correct usage
    z = arr[8];                  // Error: accessing out-of-bounds index

    // More parameter mismatches
    result = multiply();             // Error: missing both parameters
    result = add(3, 5);              // Correct use

    // Variable redeclaration in the same scope
    x = 5;
    x = 10;                      // Error: redeclaration of variable 'x' in the same scope

    // Control structures
    if (a + b) {                     // Error: non-boolean expression in 'if' condition
        print_str("Control test");
    }
    while (result) {                 // Error: non-boolean in 'while' loop condition
        result = result - 1;
    }
    return 1;
}
