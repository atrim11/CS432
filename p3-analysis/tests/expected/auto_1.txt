Invalid number of function arguments on line 28
Type mismatch in parameter 1 of call to 'multiply': expected int but found str on line 29
Invalid number of function arguments on line 30
Type mismatch in parameter 1 of call to 'add': expected int but found str on line 31
Type mismatch in parameter 0 of call to 'factorial': expected int but found str on line 39
Invalid number of function arguments on line 52
Symbol 'x' undefined on line 56
Symbol 'x' undefined on line 57
Type mismatch: bool expected but int found on line 60
Type mismatch: bool expected but int found on line 63
